challenge-manager/challenge-tree/tree.go

```diff
@@ -113,3 +114,114 @@ func (ht *RoyalChallengeTree) IsUnrivaledAtBlockNum(edge protocol.ReadOnlyEdge,
 func (ht *RoyalChallengeTree) TimeUnrivaled(edge protocol.ReadOnlyEdge, blockNum uint64) (uint64, error) {
 	return ht.LocalTimer(edge, blockNum)
 }
+
+func (ht *RoyalChallengeTree) UpdateInheritedTimer(
+	ctx context.Context,
+	edgeId protocol.EdgeId,
+	blockNum uint64,
+) (uint64, error) {
+	edge, ok := ht.edges.TryGet(edgeId)
+	if !ok {
+		return 0, fmt.Errorf("edge with id %#x not found", edgeId.Hash)
+	}
+	status, err := edge.Status(ctx)
+	if err != nil {
+		return 0, err
+	}
+	if isOneStepProven(ctx, edge, status) {
+		return math.MaxUint64, nil
+	}
+	timeUnrivaled, err := ht.TimeUnrivaled(edge, blockNum)
+	if err != nil {
+		return 0, err
+	}
+	inheritedTimer := timeUnrivaled
+
+	chalManager, err := ht.metadataReader.SpecChallengeManager(ctx)
+	if err != nil {
+		return 0, err
+	}
+	// If an edge has children, we use the min of its children if it
+	// is not a root edge. Otherwise, use the max.
```

Comment:
Sorry if its obvious but I am not sure where are we using the max of the children here?